go/pserver/etcd_lister.go

```diff
@@ -0,0 +1,120 @@
+package pserver
+
+import (
+	"time"
+	"context"
+	"strconv"
+	"strings"
+
+	"github.com/coreos/etcd/clientv3"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	DefaultEtcdTimeout time.Duration = time.Second * time.Duration(5)
+)
+
+type pserverEtcdLister struct {
```

Comment:
The name could be more concise if it's something like `etcdLister`, the package is already `pserver`, people read it in their minds as: pserver etcd lister. If it's named as `pserverEtcdLister`, it's long, and does not read good: pserver pserver etcd lister.